web-api/compose.single-machine-deploy.yml

```
@@ -0,0 +1,143 @@
+
+  elasticsearch:
+    # Don't forget to apply mapping.json to elasticsearch
+    # See example mapping in web-api-db-ctrl
+    # Make sure vector dimensions match model and run something like
+    # curl -XPUT 'http://localhost:3092/xayn/_mapping' -H 'Content-Type: application/json' -d @mapping.json
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.8.0
+    mem_limit: 8g
+    cpus: 8
+    restart: always
+    environment:
+      - discovery.type=single-node
+      - xpack.security.enabled=false
+    volumes:
+      - elasticsearch_data:/usr/share/elasticsearch/data
+    ports:
+      - "${HOST_PORT_SCOPE:-30}92:9200"
+    networks:
+      - internal
+    healthcheck:
+      test: curl --fail 'http://localhost:9200/_cluster/health'
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+  backoffice:
+    image: xaynetci/xayn_discovery_ingestion_service:main-230831084031-f320e8c-multilingual_e5_small_v0001-amd64
+    mem_limit: 1.5g
+    cpus: 2
+    restart: always
+    environment:
+      XAYN_WEB_API__STORAGE__POSTGRES__BASE_URL:  "postgres://${POSTGRES_USER:-user}:${POSTGRES_PASSWORD:-pw}@postgres:5432/xayn"
+      XAYN_WEB_API__STORAGE__ELASTIC__URL: "http://elasticsearch:9200/"
+      XAYN_WEB_API__NET__BIND_TO: "0.0.0.0:8000"
+      XAYN_WEB_API__STORAGE__POSTGRES__USER: "${POSTGRES_USER:-user}"
+      XAYN_WEB_API__STORAGE__POSTGRES__PASSWORD: "${POSTGRES_PASSWORD:-pw}"
+      XAYN_WEB_API__STORAGE__POSTGRES__APPLICATION_NAME: "xayn"
+      XAYN_WEB_API__EMBEDDING__TOKEN_SIZE: "384"
+      XAYN_WEB_API__STORAGE__ELASTIC__USER: "elastic"
+      XAYN_WEB_API__EMBEDDING__TYPE: "pipeline"
+      XAYN_WEB_API__STORAGE__ELASTIC__INDEX_NAME: "xayn"
+    ports:
+      - "${HOST_PORT_SCOPE:-30}81:8000"
+    networks:
+      - internal
+    depends_on:
+      db_ready:
+        condition: service_completed_successfully
+
+  frontoffice:
+    image: xaynetci/xayn_discovery_web_service:main-230831084031-f320e8c-multilingual_e5_small_v0001-amd64
+    mem_limit: 1.5g
+    cpus: 2
+    restart: always
+    environment:
+      XAYN_WEB_API__STORAGE__POSTGRES__BASE_URL:  "postgres://${POSTGRES_USER:-user}:${POSTGRES_PASSWORD:-pw}@postgres:5432/xayn"
+      XAYN_WEB_API__STORAGE__ELASTIC__URL: "http://elasticsearch:9200/"
+      XAYN_WEB_API__NET__BIND_TO: "0.0.0.0:8000"
+      XAYN_WEB_API__STORAGE__POSTGRES__USER: "${POSTGRES_USER:-user}"
+      XAYN_WEB_API__STORAGE__POSTGRES__PASSWORD: "${POSTGRES_PASSWORD:-pw}"
+      XAYN_WEB_API__STORAGE__POSTGRES__APPLICATION_NAME: "xayn"
+      XAYN_WEB_API__EMBEDDING__TOKEN_SIZE: "384"
+      XAYN_WEB_API__STORAGE__ELASTIC__USER: "elastic"
+      XAYN_WEB_API__EMBEDDING__TYPE: "pipeline"
+      XAYN_WEB_API__STORAGE__ELASTIC__INDEX_NAME: "xayn"
+    ports:
+      - "${HOST_PORT_SCOPE:-30}82:8000"
+    networks:
+      - internal
+    depends_on:
+      db_ready:
+        condition: service_completed_successfully
+
+  tika: 
+    image: apache/tika:2.8.0.0-full 
+    mem_limit: 256m 
```

```suggestion
    mem_limit: 512m 
```
with 256m the service fails to start on my machine